tests/unit_tests/rigid_pendulum_poc/test_quaternions.cpp

```diff
@@ -379,4 +380,99 @@ TEST(QuaternionTest, ExpectErrorWhenRotatingVectorWithNonUnitQuaternion) {
     ASSERT_THROW(rotate_vector(q, v), std::invalid_argument);
 }
 
+class QuaternionTest : public ::testing::TestWithParam<std::tuple<Quaternion, RotationMatrix>> {};
+
+TEST_P(QuaternionTest, ConvertQuaternionToRotationMatrix) {
+    auto q = std::get<0>(GetParam());  // Provided quaternion
+    auto R = std::get<1>(GetParam());  // Expected rotation matrix
+
+    // Convert quaternion to rotation matrix and compare to expected rotation matrix
+    auto R_from_q = quaternion_to_rotation_matrix(q);
+
+    EXPECT_NEAR(std::get<0>(R_from_q).GetXComponent(), std::get<0>(R).GetXComponent(), 1e-6);
+    EXPECT_NEAR(std::get<0>(R_from_q).GetYComponent(), std::get<0>(R).GetYComponent(), 1e-6);
+    EXPECT_NEAR(std::get<0>(R_from_q).GetZComponent(), std::get<0>(R).GetZComponent(), 1e-6);
+
+    EXPECT_NEAR(std::get<1>(R_from_q).GetXComponent(), std::get<1>(R).GetXComponent(), 1e-6);
+    EXPECT_NEAR(std::get<1>(R_from_q).GetYComponent(), std::get<1>(R).GetYComponent(), 1e-6);
+    EXPECT_NEAR(std::get<1>(R_from_q).GetZComponent(), std::get<1>(R).GetZComponent(), 1e-6);
+
+    EXPECT_NEAR(std::get<2>(R_from_q).GetXComponent(), std::get<2>(R).GetXComponent(), 1e-6);
+    EXPECT_NEAR(std::get<2>(R_from_q).GetYComponent(), std::get<2>(R).GetYComponent(), 1e-6);
+    EXPECT_NEAR(std::get<2>(R_from_q).GetZComponent(), std::get<2>(R).GetZComponent(), 1e-6);
+
+    // Rotate Vector {1., 1., 1.} by both quaternion and rotation matrix and compare results
+    Vector v{1., 1., 1.};
+    Vector rotated_by_q = rotate_vector(q, v);
+    Vector rotated_by_R = multiply_rotation_matrix_with_vector(R_from_q, v);
+
+    EXPECT_NEAR(rotated_by_q.GetXComponent(), rotated_by_R.GetXComponent(), 1e-6);
+    EXPECT_NEAR(rotated_by_q.GetYComponent(), rotated_by_R.GetYComponent(), 1e-6);
+    EXPECT_NEAR(rotated_by_q.GetZComponent(), rotated_by_R.GetZComponent(), 1e-6);
+
+    // Convert rotation matrix back to quaternion and compare with original
+    auto q_from_R = rotation_matrix_to_quaternion(R);
+
+    EXPECT_NEAR(q_from_R.GetScalarComponent(), q.GetScalarComponent(), 1e-6);
+    EXPECT_NEAR(q_from_R.GetXComponent(), q.GetXComponent(), 1e-6);
+    EXPECT_NEAR(q_from_R.GetYComponent(), q.GetYComponent(), 1e-6);
+    EXPECT_NEAR(q_from_R.GetZComponent(), q.GetZComponent(), 1e-6);
+}
```

Comment:
This looks like 3 tests to me

Reply:
Thanks for the suggestion, it makes sense to me. Broke up this (and the other set you mentioned) into 3 separate tests each.